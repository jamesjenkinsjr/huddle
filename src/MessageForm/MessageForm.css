.portal__message-form {
  display: flex;
  flex-direction: column;
}

.portal__message-form > label {
  display: flex;
  flex-direction: column;
}

.portal__message-form > button {
  background-color: blue;
  color: white;
}