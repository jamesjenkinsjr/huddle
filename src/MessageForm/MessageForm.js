import React from 'react'
import './MessageForm.css'
import MessageService from '../services/message-api-service'

export default class MessageForm extends React.Component {
  state = {
    error: '',
  }

  handleSubmit = e => {
    e.preventDefault()
    const { author, content } = e.target
    const newMessage = { author: author.value, content: content.value }
    newMessage.portal_id = this.props.portal_id
    MessageService.addMessage(newMessage)
      .then((message) => {
        MessageService.getMessageByID(message.id)
        this.props.handleNewMessage(message)
        content.value = ''
      })
      .catch(error => {
        this.setState({
          error: error.message,
        })
      })
  }

  handleTextarea = e => {
    if(!e.shiftKey && e.key === 'Enter') {
      e.preventDefault()
      e.stopPropagation()
      this.submitButton.click()
    }
  }

  render() {
    return (
      <form
        className="portal__message-form"
        onSubmit={this.handleSubmit}
      >
        <label htmlFor="author">
          Name
          <input type="text" name="author" id="author" />
        </label>
        <label htmlFor="content">
          Message
          <textarea
            onKeyPress={this.handleTextarea}
            name="content"
            id="content"
            required
          ></textarea>
        </label>
        <button type="submit" ref={input => this.submitButton = input}>Submit</button>
      </form>
    )
  }
}
