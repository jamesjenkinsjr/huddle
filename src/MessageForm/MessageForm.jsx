import React from 'react'
import { withCookies } from 'react-cookie'
import './MessageForm.css'
import MessageService from '../services/message-api-service'

class MessageForm extends React.Component {
  state = {
    error: '',
  }

  handleSubmit = e => {
    e.preventDefault()
    const { author, content } = e.target
    const newMessage = { author: author.value, content: content.value }
    newMessage.portal_id = this.props.portal_id
    MessageService.addMessage(newMessage)
      .then(message => {
        MessageService.getMessageByID(message.portal_id, message.id)
        this.props.handleNewMessage(message)
        content.value = ''
      })
      .catch(error => {
        this.setState({
          error: error.message,
        })
      })
  }

  handleTextarea = e => {
    if (!e.shiftKey && e.key === 'Enter') {
      e.preventDefault()
      e.stopPropagation()
      this.submitButton.click()
    }
  }

  setNameCookie = e => {
    e.preventDefault()
    this.props.cookies.set(`name-${this.props.portal_id}`, e.target.value, {path: `/${this.props.portal_id}`})
  }

  getNameCookie = () => {
    const nameCookie = this.props.cookies.get(`name-${this.props.portal_id}`)
    return nameCookie ? nameCookie : ''
  }

  render() {
    return (
      <form className="portal__message-form" onSubmit={this.handleSubmit}>
        <label htmlFor="author">
          Name
          <input type="text" name="author" id="author" defaultValue={this.getNameCookie()} onChange={this.setNameCookie} />
        </label>
        <label htmlFor="content">
          Message
          <textarea
            onKeyPress={this.handleTextarea}
            name="content"
            id="content"
            required
          ></textarea>
        </label>
        <button type="submit" ref={input => (this.submitButton = input)}>
          Submit
        </button>
      </form>
    )
  }
}

export default withCookies(MessageForm)
