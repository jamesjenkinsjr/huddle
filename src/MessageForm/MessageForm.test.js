import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import MessageForm from './MessageForm'

describe('MessageForm component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<MessageForm />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
