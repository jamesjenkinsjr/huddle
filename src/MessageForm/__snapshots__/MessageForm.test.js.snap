// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm component renders without crashing 1`] = `
<form
  className="portal__message-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="author"
  >
    Name
    <input
      id="author"
      name="author"
      type="text"
    />
  </label>
  <label
    htmlFor="content"
  >
    Message
    <textarea
      id="content"
      name="content"
      onKeyPress={[Function]}
      required={true}
    />
  </label>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
